libraries/entities/test/entity_scan_test.js

```diff
@@ -302,4 +318,114 @@ helper.dbSuite(path.basename(__filename), function() {
       assert.equal(result.entries.length, 0);
     });
   });
+  suite('scan SQL injection checks', function() {
+    const properties = {
+      taskId: Entity.types.String,
+      provisionerId: Entity.types.String,
+      workerType: Entity.types.String,
+      count: Entity.types.Number,
+      expires: Entity.types.Date,
+    };
+    const configuredTestTable = Entity.configure({
+      version: 1,
+      partitionKey: Entity.keys.StringKey('taskId'),
+      rowKey: Entity.keys.StringKey('provisionerId'),
+      properties,
+    });
+
+    test('retrieve documents with string condition', async function() {
+      db = await helper.withDb({ schema, serviceName });
+      const TestTable = configuredTestTable.setup({ tableName: 'test_entities', db, serviceName });
+      await assert.rejects(
+        () => TestTable.scan({workerType: Entity.op.equal("foo")}),
+        /can only be a date/);
+    });
+
+    test('retrieve documents with SQL escape in partition key', async function() {
+      db = await helper.withDb({ schema, serviceName });
+      const TestTable = configuredTestTable.setup({ tableName: 'test_entities', db, serviceName });
+      const document = {
+        taskId: '1',
+        provisionerId: 'test',
+        workerType: 'test',
+        count: 13,
+        expies: new Date(),
+      };
+      await TestTable.create({ ...document, taskId: '1', expires: new Date('2020-01-01') });
+      // Enabling EXPLAIN for this test shows (newlines added):
+      // Query Text:
+      //   select
+      //     test_entities.partition_key,
+      //     test_entities.row_key,
+      //     test_entities.value,
+      //     test_entities.version,
+      //     test_entities.etag
+      //   from test_entities
+      //   where partition_key = ''' or 1=1; drop table test_entities; --'
+      //   order by test_entities.partition_key, test_entities.row_key
+      //   limit 1000 offset 0
+      //
+      // Which contains a correct quoting of the `'` character:
+      //
+      // postgres=# select ''' or 1=1; drop table test_entities; --';
+      // ----------------------------------------
+      //  ' or 1=1; drop table test_entities; --
+      //
+      // Furthermore, replacing the usage of `quote_literal` with a vulnerable equivalent results
+      // in the following from this test, indicating a successful injection:
+      //
+      //   error: cannot open multi-query plan as cursor
+      //   WHERE: PL/pgSQL function test_entities_scan(text,text,text,integer,integer) line 31 at RETURN QUERY
+      //   CODE: 42P11
+      const res = await TestTable.scan({taskId: "' or 1=1; drop table test_entities; --"});
```

Comment:
What does `--` do?

Reply:
It's a comment.  I got that string from some "hax0r your way into websites!" guide.  I think the idea is just to comment out any additional SQL that the server might add after the user input.